lib/rmg/Zone.h

```diff
@@ -119,7 +119,7 @@ class Zone : public rmg::ZoneOptions
 	template<class T>
 	void addModificator()
 	{
-		modificators.emplace_back(new T(*this, map, generator));
+		modificators.push_back(std::make_unique<T>(*this, map, generator));
```

Comment:
Why is this problem?

Reply:
I think it's related to the usage of naked `new` which is usually not a good idea in modern C++

Reply:
I like to use new because VS intellisence works better with it than with std::make_unique/std::make_shared. So ptr.reset(new Something()) works better for me

Reply:
> I like to use new because VS intellisence works better with it than with std::make_unique/std::make_shared. So ptr.reset(new Something()) works better for me

well, it's a problem of IDE and is not an excuse to avoid good code. Install Visual Assist, life will become much better ;)

Reply:
It's not naked new, it emplaces unique_ptr. Change here doesn't fix any bug or memory leak, it just trigger for endless holy war discussion